### ISSUE TITLE
Duckling keeps returning old date data even if the text doesn't contain it anymore

### ISSUE THREAD

**Issue by malave** [samtecspg/articulate#133]
So I tried this text and duckling correctly interpreted the from/to dates:

`Flights from Miami to San Jose on May 22 to June 11`

and returns:

```
...
 "duckling_time": {
            "from": "2018-05-22T00:00:00.000-04:00",
            "to": "2018-06-12T00:00:00.000-04:00",
            "original": "on May 22 to June 11"
          }
...
```

But if after that I send the text: 

`Flights from Miami to San Jose`

it keeps returning the same time object until a change the text and add a new date

**Comment by ewoo** [comment]
@wrathagom I agree. That's how it should behave by default. @malave to reset this, the caller is expected to pass in a new set of dates.

I could see two ways to improve this--for the future.
1. create a magic word to release this property in the context object (not sure how)
2. provide a property in the slots to persist it in the context object--or require Articulate to clarify saved context on each subsequent request, "for the same date periods?" (Could get annoying)

**Comment by malave** [comment]
So I have a few scenarios in my hackaton project, a flight quotes bot, that the current duckling behavior doesn't work.

### Auto-fills the `to` value with `from` + 1 day
When I try to get quotes for different days to check better flight prices. I could ask something like: 

> Flights from Miami to Madrid on May 11 to May 18

Duckling will correctly get the `from` and `to` dates. 
But if after that I want to ask something like: 

> Flights from Miami to Madrid on May 11  

( I do this because sometimes round trips are cheaper than one-way). 

In this case duckling will correctly get the `from` but the `to` value will be `May 12`, in this case I would expect a `[null|undefined|empty]` value so I can tell the API to use the `anytime` keyword instead of an specific date.

I haven't work too much with duckling so I don't know if this is configurable or not.

### Need to check for a null slot but instead get session values
Another scenario is asking for specific dates and then asking for a recommendation from the API service. 

> Flights from Miami to Madrid on May 11 to May 18

Then ask:

> Flights from Miami to Madrid

The API either request to use the `anytime` in both `from` and `to` dates or just not use them at all and it will return quotes with recommendations. In this case I would like to check if there are no dates then don't use them in the API call, but instead I will receive the dates from the previous query (`May 11 to May 18`). 

Probably not a lot of people will do something like this when searching for flight information but in general I can see  projects that you will need to check if something is null to proceed in a different way.

Maybe there should be a way to identify session data from data from the last conversation, that way the developer will decide if is relevant or not at that moment.

**Comment by wrathagom** [comment]
@malave how much of this do you think the user would encounter vs is testing related. Part of me thinks that truly distinct searches would be on different session ids and thus not get stuck within the same context. and that within a single session it's probably not a bad idea to assume that the suer is searching for the same dates.

**Comment by malave** [comment]
@wrathagom If I'm testing different dates to see how prices change I would like to try an starting date and get recommendations on the return day. In this scenario I will get back the initial day +1 on the `to` field.

**Comment by wrathagom** [comment]
Would be addressed by #154 closing in favor of that issue.
